### ISSUE TITLE
Demon's Souls Pixelated/Blocky Depth of Field With Resolutions Above 720p

### ISSUE THREAD

**Issue by auwsmit** [RPCS3/rpcs3#8475]
## Quick summary
At resolutions higher than 720, the depth of field effect becomes blocky/pixelated, and the effect is more noticeable the higher the resolution.

## Details
At 1080, the blur becomes subtly pixelated. And at 1440 or higher, it's very noticeable.

#### 1. Please attach RPCS3's log.
Figured I don't need a log since this seems to happen 100% of the time that there are 3D graphics on screen. It's noticeable right at the start of the tutorial (and throughout the game) on the edges of buildings/structures in the distance, and very blatant when looking out at distant views/vistas.

#### 2. If you describe graphical issue, please provide an RSX capture and a RenderDoc capture that demonstrate it.
[RSX capture](https://mega.nz/file/nwY2TCCT#oJpeJpn7pKxFOr_gdKxflaOU79oQ_xrW5HDgbhOKRIE)
[RenderDoc capture](https://mega.nz/file/HlJUyajQ#5PaSbDmsI6wE6yvSTJFgIDPas32VGlwC-mCqKhV-VKI)

#### 3. Please attach screenshots of your problem.
![2020-06-21 03_02_37-Window](https://user-images.githubusercontent.com/6760531/85221816-d3880200-b36b-11ea-9634-035d449d71e1.png)

#### 4. Please provide comparison with real PS3.
The depth of field effect should look like a perfectly smooth blur. It looks accurate when the emulator is set to the original resolution of 720.

#### 5. Please provide your system configuration:
* Windows 10
* Ryzen 5 3600
* RTX 2070 Super
* 445.87

#### Please include.
I can imagine this might be similar to asking for higher resolution textures, since the game was never intended to run at higher resolutions, so this won't be considered a bug and won't be fixed.

However, if you can figure out the root cause and point me in the right direction for fixing it, or possibly making some sort of patch/hotfix for it, that would be very appreciated. So far this has been the only flaw in my 60 fps 1440p playthrough, which has been otherwise awesome. It's also worth pointing out that there's a [relatively popular mod on Nexus Mods with a few thousand downloads that removes the depth of field effect](https://www.nexusmods.com/demonssouls/mods/3) likely for this reason (granted many people might just dislike DoF anyway, lol).

**Comment by Emogop** [comment]
It's really not an issue.

**Comment by auwsmit** [comment]
It's not detrimental to gameplay in any way, but it's very distracting for me personally. A huge part of my appreciation of these games is taking in the atmosphere and environment, so it's hard to not notice the wonky edges of buildings and blocky views.

**Comment by Asinin3** [comment]
Resolution scaling is technically a hack though. And I wouldn't be surprised if you can already fix this by increasing the resolution scaling threshold. Issues like this generally get closed

**Comment by auwsmit** [comment]
I thought that might fix it too, and I swear I tried messing with it to no avail before making this post.

But after being more rigorous and trying many values, I've found that setting the resolution scaling threshold to 640x640 or higher seems to do the trick, at least for 1440p resolution:

![2020-06-21 18_04_24-Window](https://user-images.githubusercontent.com/6760531/85239698-d1fd1f00-b3e9-11ea-8237-a9b32c7dc045.png)

Thanks for pointing me in the right direction! When I was a kid I remember Gamecube emulation being unplayably slow on pretty great hardware, and now we have a PS3 emulator that can play some games not only at full speed, but at an improved framerate and resolution. It's like magic.
